tests/unit/restapi/test_tags_v1.py

```diff
@@ -0,0 +1,525 @@
+    """
+
+    return client.delete(
+        f"/{V1_ROOT}/{V1_TAGS_ROUTE}/{tag_id}",
+        follow_redirects=True,
+    )
+
+
+# -- Assertions ------------------------------------------------------------------------
+
+
+def assert_tag_response_contents_matches_expectations(
+    response: dict[str, Any], expected_contents: dict[str, Any]
+) -> None:
+    expected_keys = {
+        "id",
+        "group",
+        "user",
+        "createdOn",
+        "lastModifiedOn",
+        "name",
+    }
+    assert set(response.keys()) == expected_keys
+
+    # Validate the non-Ref fields
+    assert isinstance(response["id"], int)
+    assert isinstance(response["name"], str)
+    assert isinstance(response["createdOn"], str)
+    assert isinstance(response["lastModifiedOn"], str)
+
+    assert response["name"] == expected_contents["name"]
+
+    assert is_iso_format(response["createdOn"])
+    assert is_iso_format(response["lastModifiedOn"])
+
+    # Validate the UserRef structure
+    assert isinstance(response["user"]["id"], int)
+    assert isinstance(response["user"]["username"], str)
+    assert isinstance(response["user"]["url"], str)
+    assert response["user"]["id"] == expected_contents["user_id"]
+
+    # Validate the GroupRef structure
+    assert isinstance(response["group"]["id"], int)
+    assert isinstance(response["group"]["name"], str)
+    assert isinstance(response["group"]["url"], str)
+    assert response["group"]["id"] == expected_contents["group_id"]
+
+
+def assert_retrieving_tag_by_id_works(
+    client: FlaskClient,
+    tag_id: int,
+    expected: dict[str, Any],
+) -> None:
+    """Assert that retrieving a tag by id works.
+
+    Args:
+        client: The Flask test client.
+        tag_id: The id of the tag to retrieve.
+        expected: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_TAGS_ROUTE}/{tag_id}", follow_redirects=True
+    )
+    assert response.status_code == 200 and response.get_json() == expected
+
+
+# TODO handle Q,G,P
+def assert_retrieving_all_tags_works(
+    client: FlaskClient,
+    expected: list[dict[str, Any]],
+    Q: dict[str, Any] | None = None,
+    G: dict[str, Any] | None = None,
+    P: dict[str, Any] | None = None,
+) -> None:
+    """Assert that retrieving all tags works.
+
+    Args:
+        client: The Flask test client.
+        expected: The expected response from the API.
+        Q: The search tag parameters.
+        G: The group tag parameters.
+        P: The paging tag parameters.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+
+    payload = {}
+
+    if Q:
+        payload.update(Q)
+    if G:
+        payload.update(G)
+    if P:
+        payload.update(P)
```

Comment:
Query parameters and JSON payloads need to be separated out, see requested changes in #447 for more details.